cmd/wick/helpers.go

```diff
@@ -60,55 +71,68 @@ func selectAuthMethod(privateKey string, ticket string, secret string) string {
 	return "anonymous"
 }
 
-func userHomeDir() string {
-	if runtime.GOOS == "windows" {
-		home := os.Getenv("HOMEDRIVE") + os.Getenv("HOMEPATH")
-		if home == "" {
-			home = os.Getenv("USERPROFILE")
-		}
-		return home
-	} else if runtime.GOOS == "linux" {
-		home := os.Getenv("XDG_CONFIG_HOME")
-		if home != "" {
-			return home
-		}
+func validateData(sessionCount int, concurrency int, keepAlive int) error {
+	if sessionCount < 1 {
+		return fmt.Errorf("parallel must be greater than zero")
+	}
+	if concurrency < 1 {
+		return fmt.Errorf("concurrency must be greater than zero")
 	}
-	return os.Getenv("HOME")
+	if keepAlive < 0 {
+		return fmt.Errorf("keepalive interval must be greater than zero")
+	}
+
+	return nil
 }
 
-func readFromProfile() {
-	cfg, err := ini.Load(fmt.Sprintf("%s/.wick/config", userHomeDir()))
+func readFromProfile(profile string) (*core.ClientInfo, error) {
+	clientInfo := &core.ClientInfo{}
+	cfg, err := ini.Load(os.ExpandEnv("$HOME/.wick/config"))
 	if err != nil {
-		log.Fatalf("Fail to read config: %v", err)
+		return nil, fmt.Errorf("fail to read config: %v", err)
 	}
 
-	section, err := cfg.GetSection(*profile)
+	section, err := cfg.GetSection(profile)
 	if err != nil {
-		log.Fatalf("Error in getting section: %s", err)
+		return nil, fmt.Errorf("error in getting section: %s", err)
 	}
 
-	*url = section.Key("url").Validate(func(s string) string {
+	clientInfo.Url = section.Key("url").Validate(func(s string) string {
 		if len(s) == 0 {
 			return "ws://localhost:8080/ws"
 		}
 		return s
 	})
-	*realm = section.Key("realm").Validate(func(s string) string {
+	clientInfo.Realm = section.Key("realm").Validate(func(s string) string {
 		if len(s) == 0 {
 			return "realm1"
 		}
 		return s
 	})
-	*authid = section.Key("authid").String()
-	*authrole = section.Key("authrole").String()
-	*authMethod = section.Key("authmethod").String()
-	if *authMethod == "cryptosign" {
-		*privateKey = section.Key("private-key").String()
-	} else if *authMethod == "ticket" {
-		*ticket = section.Key("ticket").String()
-	} else if *authMethod == "wampcra" {
-		*secret = section.Key("secret").String()
+	switch section.Key("serializer").String() {
+	case "msgpack", "cbor", "json", "":
+		clientInfo.Serializer = getSerializerByName(section.Key("serializer").Validate(func(s string) string {
+			if len(s) == 0 {
+				return "json"
+			}
+			return s
+		}))
+	default:
+		return nil, fmt.Errorf("serailizer must be json, msgpack or cbor")
+	}
```

Comment:
ah, there's a default, how about being a bit more explicit then?
```suggestion
	serializer := section.Key("serializer").String()
	switch serializer {
	case "msgpack", "cbor", "json":
		clientInfo.Serializer = getSerializerByName(serializer)
	case "":
		// default to json if none was provided
		clientInfo.Serializer = getSerializerByName("json")
	default:
		return nil, fmt.Errorf("serailizer must be json, msgpack or cbor")
	}
```

Reply:
Done